com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs

```diff
@@ -740,25 +740,9 @@ static void InitializeShadowData(UniversalRenderPipelineAsset settings, NativeAr
 
             // We no longer use screen space shadows in URP.
             // This change allows us to have particles & transparent objects receive shadows.
-            shadowData.requiresScreenSpaceShadowResolve = false;// shadowData.supportsMainLightShadows && supportsScreenSpaceShadows && settings.shadowCascadeOption != ShadowCascadesOption.NoCascades;
+            shadowData.requiresScreenSpaceShadowResolve = false;
 
-            int shadowCascadesCount;
-            switch (settings.shadowCascadeOption)
-            {
-                case ShadowCascadesOption.FourCascades:
-                    shadowCascadesCount = 4;
-                    break;
-
-                case ShadowCascadesOption.TwoCascades:
-                    shadowCascadesCount = 2;
-                    break;
-
-                default:
-                    shadowCascadesCount = 1;
-                    break;
-            }
-
-            shadowData.mainLightShadowCascadesCount = shadowCascadesCount;//(shadowData.requiresScreenSpaceShadowResolve) ? shadowCascadesCount : 1;
+            shadowData.mainLightShadowCascadesCount = settings.cascadeShadowSplitCount;
```

Comment:
Split Count is different from Cascade count. 4 cascades means 3 splits (Vector3). So I'm confused. Maybe the API should return cascade count if that's what you want.

Reply:
done, makes more sense now.